Examples/en/TypeScript/Language/Soundness.ts

```diff
@@ -0,0 +1,110 @@
+// Without a background in type theory, you're unlikely
```

Comment:
Another unsound feature that might be worth mentioning is this (I can't think of an example that isn't contrived right now, so maybe it's not a good inclusion!)
```ts
// We can always add an extra optional property to a type.
const obj: { x: string; y? : number } = { x: "a property "};

// However in combination with the ability to 'forget' properties this can
// can be unsound.

const objXY = { x: "a property", y: true };
const objX: { x: string } = objXY;
const objXYNumber: { x: string, y?: number } = objX;

const num: number = objXYNumber.y === undefined ? 0 : objXYNumber.y;

// Excess property checking can prevent errors like this when you
// initialise your objects with annotations:

const objXYAnnotation: { x: string } = { x: "a property", y: true }; // error
```